master/cluster.go

```diff
@@ -1002,10 +1040,24 @@ func (c *Cluster) addMetaNode(nodeAddr, zoneName string, nodesetId uint64) (id u
 		if nodesetId > 0 && nodesetId != metaNode.ID {
 			return metaNode.ID, fmt.Errorf("addr already in nodeset [%v]", nodeAddr)
 		}
+
+		// compatible with old version in which raft heartbeat port and replica port did not persist
```

Comment:
The judgment should be consistent with the condition of 'RaftPartitionCanUsingDifferentPortEnable'. Specifically, if this condition is true, nodes registered with the raft port 0 should not be allowed. When an SRE deploys a node with an old version, it might have an impact on the stability of the system.